File: question4/spec/cake_q4_spec.rb
require 'spec_helper'
require 'rinda/tuplespace'
require_relative '../lib/cake_q4.rb'

RSpec.describe 'Interview Cake Question 4' do
  let(:meeting_times_1) { [[1,3], [3,5]] }
  let(:meeting_times_2) { [[1,10],[2,6],[4,5], [7,9]] }
  let(:meeting_times_3) { [[3,5],[4,8],[10,12],[9,10],[0,1]] }

  it 'condenses the meeting times if the times are touching' do
    expect(condense_meetings(meeting_times_1)).to eq [[1,5]]
  end

  it 'condenses the meeting from earliest start time to latest stop time' do
    expect(condense_meetings(meeting_times_2)).to eq [[1,10]]
  end

  it 'condenses the meeting times even when meeting times are not in order' do
    expect(condense_meetings(meeting_times_3)).to eq [[0, 1], [3, 8], [9, 12]]
  end
end

File: question-04-merge-meeting-times/spec/cakeQ4_spec.js
import MeetingMerger from '../lib/cakeQ4'

describe('MeetingMerger', () => {

  let array1, array2, array3, array4, instance1, instance2, instance3, instance4;

  beforeEach(() => {
    array1 = [[1,3], [3,5]];
    array2 = [[1,10],[2,6],[4,5], [7,9]];
    array3 = [[3,5],[4,8],[10,12],[9,10],[0,1]];
    instance1 = new MeetingMerger(array1);
    instance2 = new MeetingMerger(array2);
    instance3 = new MeetingMerger(array3);
  });

  // assert
  it('condenses the meeting times if the times are touching', () => {
    expect(instance1.previousEnd).toEqual(3);
    expect(instance1.sorted).toEqual([[1,3], [3,5]]);
    expect(instance3.sorted).toEqual([[0,1],[3,5],[4,8],[9,10],[10,12]]);
    expect(instance1.condenseTimes()).toEqual([[1,5]]);
  });

  it('condenses the meeting from earliest start time to latest stop time', () => {});

  it('condenses the meeting times even when meeting times are not in order', () => {});

  // teardown
  afterEach(() => {
    array1 = null;
    instance1 = null;
  });
});
